frontend/javascripts/oxalis/view/action-bar/starting_job_modals.tsx

```diff
@@ -694,6 +714,54 @@ function useCurrentlySelectedBoundingBox(
   return currentlySelectedBoundingBox;
 }
 
+const isBBoxTooSmall = (
+  bbox: Vector6 | Vector3,
+  segmentationType: APIJobType.INFER_NEURONS | APIJobType.INFER_MITOCHONDRIA,
+  bboxOrDS: "bbox" | "dataset" = "bbox",
+) => {
+  const minBBoxExtent =
+    segmentationType === APIJobType.INFER_NEURONS
+      ? MIN_BBOX_EXTENT_FOR_NEURON_SEGMENTATION
+      : MIN_BBOX_EXTENT_FOR_MITO_SEGMENTATION;
+  let minExtentInMag1 = minBBoxExtent.minExtent;
+  if (!_.isEqual(minBBoxExtent.mag, [1, 1, 1]) || bboxOrDS === "dataset") {
+    const extentFactor = bboxOrDS === "dataset" ? MIN_DATASET_RATIO_COMPARED_TO_MIN_BBOX : 1;
+    minExtentInMag1 = minBBoxExtent.minExtent.map(
+      (extent, i) => extent * minBBoxExtent.mag[i] * extentFactor,
+    ) as Vector3;
+  }
+  const bboxExtent = bbox.length === 6 ? bbox.slice(3) : bbox;
+  for (let i = 0; i < 3; i++) {
+    if (bboxExtent[i] < minExtentInMag1[i]) {
+      const boundingBoxOrDSMessage = bboxOrDS === "bbox" ? "bounding box" : "dataset";
+      Toast.error(
+        `The ${boundingBoxOrDSMessage} is too small. Please select a ${boundingBoxOrDSMessage} with the minimal extent ${minExtentInMag1} Vx.`,
+      );
+      return true;
+    }
+  }
+  return false;
+};
+
+const isDatasetOrBoundingBoxTooSmall = (
+  bbox: Vector6 | Vector3,
+  colorLayer: APIDataLayer,
+  segmentationType: APIJobType.INFER_NEURONS | APIJobType.INFER_MITOCHONDRIA,
+): boolean => {
+  const datasetExtent: Vector3 = [
+    colorLayer.boundingBox.width,
+    colorLayer.boundingBox.height,
+    colorLayer.boundingBox.depth,
+  ];
+  if (isBBoxTooSmall(datasetExtent, segmentationType, "dataset")) {
+    return true;
+  }
+  if (isBBoxTooSmall(bbox, segmentationType)) {
```

Comment:
Might be personal preference, but if you build the `datasetExtent` from the `colorLayer.boundingBox` here anyways, you could do the same for `bbox` and get rid of the special casing in the `isBBoxTooSmall` function (`const bboxExtent = bbox.length === 6 ? bbox.slice(3) : bbox;`).